src/test/regress/expected/recurring_outer_join.out

```diff
@@ -0,0 +1,568 @@
+ count
+---------------------------------------------------------------------
+    28
+(1 row)
+
+SELECT COUNT(*) FROM dist_1 RIGHT JOIN ref_1 USING (a);
+DEBUG:  recursively planning left side of the right join since the outer side is a recurring rel
+DEBUG:  recursively planning distributed relation "dist_1" since it is part of a distributed join node that is outer joined with a recurring rel
+DEBUG:  Wrapping relation "dist_1" to a subquery
+DEBUG:  generating subplan XXX_1 for subquery SELECT a FROM recurring_outer_join.dist_1 WHERE true
+DEBUG:  Plan XXX query after replacing subqueries and CTEs: SELECT count(*) AS count FROM ((SELECT dist_1_1.a, NULL::integer AS b FROM (SELECT intermediate_result.a FROM read_intermediate_result('XXX_1'::text, 'binary'::citus_copy_format) intermediate_result(a integer)) dist_1_1) dist_1 RIGHT JOIN recurring_outer_join.ref_1 USING (a))
+ count
+---------------------------------------------------------------------
+    28
+(1 row)
+
+SELECT COUNT(*) FROM ref_1 FULL JOIN dist_1 USING (a);
+DEBUG:  recursively planning right side of the full join since the other side is a recurring rel
+DEBUG:  recursively planning distributed relation "dist_1" since it is part of a distributed join node that is outer joined with a recurring rel
+DEBUG:  Wrapping relation "dist_1" to a subquery
+DEBUG:  generating subplan XXX_1 for subquery SELECT a FROM recurring_outer_join.dist_1 WHERE true
+DEBUG:  Plan XXX query after replacing subqueries and CTEs: SELECT count(*) AS count FROM (recurring_outer_join.ref_1 FULL JOIN (SELECT dist_1_1.a, NULL::integer AS b FROM (SELECT intermediate_result.a FROM read_intermediate_result('XXX_1'::text, 'binary'::citus_copy_format) intermediate_result(a integer)) dist_1_1) dist_1 USING (a))
+ count
+---------------------------------------------------------------------
+    36
+(1 row)
+
+SELECT COUNT(*) FROM dist_1 FULL JOIN ref_1 USING (a);
+DEBUG:  recursively planning left side of the full join since the other side is a recurring rel
+DEBUG:  recursively planning distributed relation "dist_1" since it is part of a distributed join node that is outer joined with a recurring rel
+DEBUG:  Wrapping relation "dist_1" to a subquery
+DEBUG:  generating subplan XXX_1 for subquery SELECT a FROM recurring_outer_join.dist_1 WHERE true
+DEBUG:  Plan XXX query after replacing subqueries and CTEs: SELECT count(*) AS count FROM ((SELECT dist_1_1.a, NULL::integer AS b FROM (SELECT intermediate_result.a FROM read_intermediate_result('XXX_1'::text, 'binary'::citus_copy_format) intermediate_result(a integer)) dist_1_1) dist_1 FULL JOIN recurring_outer_join.ref_1 USING (a))
+ count
+---------------------------------------------------------------------
+    36
+(1 row)
+
+-- distributed side is a subquery
+SELECT COUNT(*) FROM ref_1 LEFT JOIN (SELECT * FROM dist_1) q USING (a);
```

Comment:
sometimes Postgres pulls-up very simple queries, so it is worth adding another test with:

SELECT COUNT(*) FROM ref_1 LEFT JOIN (SELECT *,random() FROM dist_1) q USING (a);

see pull_up_subqueries / is_simple_subquery for the details